src/main/java/com/lorepo/icplayer/client/module/NestedAddonUtils.java

```diff
@@ -0,0 +1,57 @@
+package com.lorepo.icplayer.client.module;
+
+import com.google.gwt.dom.client.Element;
+import com.google.gwt.user.client.ui.HTML;
+
+public class NestedAddonUtils {
+
+    public static String CLASS_NAME = "ic_addon_gap";
+    public static String ID_PREFIX = "addonGap-";
+    public static String ID_EDITOR_PREFIX = "editor-addonGap-";
+    public static String GAP_PREFIX = "\\addon{";
+
+    public static String getDefinition(String addonId) {
+        return GAP_PREFIX + addonId + "}";
+    }
+
+    public static Element getPlaceholder(String addonId) {
+        HTML html = new HTML("<input></input>");
+        Element el = html.getElement().getFirstChildElement();
+        updatePlaceholder(addonId, el);
+        return el;
+    }
+
+    public static Element getPlaceholderInEditorMode(String addonId) {
+        HTML html = new HTML("<input></input>");
+        Element el = html.getElement().getFirstChildElement();
+        updatePlaceholderInEditorMode(addonId, el);
+        return el;
+    }
+
+    public static void updatePlaceholder(String addonId, Element el) {
+        el.addClassName(CLASS_NAME);
+        el.setId(ID_PREFIX + addonId);
+        el.setAttribute("placeholder", addonId);
+        el.setAttribute("size", Integer.toString(addonId.length() + 1));
+    }
+
+    public static void updatePlaceholderInEditorMode(String addonId, Element el) {
+        updatePlaceholder(addonId, el);
+        el.setAttribute("data-addon-value", getDefinition(addonId));
+    }
+
+
+    public static native boolean insertIntoAddonGap(String moduleID, Element view, Element page) /*-{
```

Comment:
Nie wiem czy nie rozbiłbym tego na dwie funkcje - pierwsza canInsertIntoAddonGap a druga insertIntoAddonGap. To tylko sugestia.

Reply:
Bo w przypadku kiedy addonGap.lenth == 0 to addon nie zostanie insertowany a sama nazwa nie sugeruje nam, że dostaniemy false/true (np. nazwa canXXX, isXXX, didXXX).